js/ui/form.d.ts

```diff
@@ -40,6 +47,44 @@ import Widget, {
     WidgetOptions
 } from './widget/ui.widget';
 
+/** @public */
+export type ContentReadyEvent = ComponentEvent<dxForm>;
+
+/** @public */
+export type DisposingEvent = ComponentDisposingEvent<dxForm>;
+
+/** @public */
+export type EditorEnterKeyEvent = ComponentEvent<dxForm> & {
+    readonly dataField?: string;
+}
+
+/** @public */
+export type FieldDataChangedEvent = ComponentEvent<dxForm> & {
+    readonly dataField?: string;
+    readonly value?: any;
+}
+
+/** @public */
+export type InitializedEvent = ComponentInitializedEvent<dxForm>;
+
+/** @public */
+export type OptionChangedEvent = ComponentOptionChangedEvent<dxForm>;
+
+/** @public */
+export type GroupItemTemplateData = {
```

Comment:
@IgnatovDan, could you review `GroupItemTemplateData` and `SimpleItemTemplateData`? I mean readonly and optional modifiers

Reply:
"component" should be passed
"formData" can be null/undefined
https://github.com/DevExpress/DevExtreme/blob/21_1/js/ui/form/ui.form.js#L567
https://js.devexpress.com/Documentation/21_1/ApiReference/UI_Components/dxForm/Item_Types/GroupItem/#template

Reply:
SimpleItemTemplateData:

"component" is not null, readonly
other fields can be null/undefined, readonly

https://github.com/DevExpress/DevExtreme/blob/21_1/js/ui/form/ui.form.layout_manager.js#L924 - too complex code to review each property and no tests.
https://js.devexpress.com/Documentation/21_1/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/#template

Reply:
Great, thank you